tests/CMakeLists.txt

```diff
@@ -219,49 +105,16 @@ function(add_test_suite suite_name)
         string(SUBSTRING ${generated_data_name} 11 -1 generated_data_name)
         list(APPEND config_generated_data_names ${generated_data_name})
     endforeach()
-    foreach(generated_data_file ${ecp_generated_data_files})
-        # Get the plain filename
-        get_filename_component(generated_data_name ${generated_data_file} NAME)
-        # Remove the ".data" extension
-        get_name_without_last_ext(generated_data_name ${generated_data_name})
-        # Remove leading "test_suite_"
-        string(SUBSTRING ${generated_data_name} 11 -1 generated_data_name)
-        list(APPEND ecp_generated_data_names ${generated_data_name})
-    endforeach()
-    foreach(generated_data_file ${psa_generated_data_files})
-        # Get the plain filename
-        get_filename_component(generated_data_name ${generated_data_file} NAME)
-        # Remove the ".data" extension
-        get_name_without_last_ext(generated_data_name ${generated_data_name})
-        # Remove leading "test_suite_"
-        string(SUBSTRING ${generated_data_name} 11 -1 generated_data_name)
-        list(APPEND psa_generated_data_names ${generated_data_name})
-    endforeach()
 
-    if(";${bignum_generated_data_names};" MATCHES ";${data_name};")
-        set(data_file
-            ${CMAKE_CURRENT_BINARY_DIR}/suites/test_suite_${data_name}.data)
-        set(dependency test_suite_bignum_generated_data)
-    elseif(";${config_generated_data_names};" MATCHES ";${data_name};")
+    if(";${config_generated_data_names};" MATCHES ";${data_name};")
```

Comment:
The path to the data file is based on CMAKE_CURRENT_BINARY_DIR in one case, CMAKE_CURRENT_SOURCE_DIR otherwise thus that's one difference. Otherwise, the `set(dependency test_suite_config_generated_data)` can be moved out of the if-else.

Reply:
Ah I did indeed miss something, thanks!